backend/app/src/main/java/com/moh/phlat/backend/controller/ProcessDataController.java

```diff
@@ -64,26 +74,45 @@ public class ProcessDataController {
 	@GetMapping("/view/all")
 	public @ResponseBody ResponseEntity<ResponseMessage> getAllProcessDatas() {
 		return ResponseEntity.status(HttpStatus.OK)
-				.body(new ResponseMessage("success", 200, "", processDataRepository.findAll()));
+				.body(new ResponseMessage("success", 200, "", null, processDataRepository.findAll()));
 	}
 
 	// get process data by control id
 	@PreAuthorize("hasAnyRole(@roleService.getAllRoles())")
 	@PostMapping("/controltable/{controlTableId}")
 	public @ResponseBody ResponseEntity<ResponseMessage> getAllProcessDataByControlTableId(
-		@PathVariable Long controlTableId, @RequestParam(required = false) String rowStatus, @RequestBody ProcessDataFilterParams filterProcess) {
+			@PathVariable Long controlTableId, @RequestParam(required = false) String rowStatus, @RequestParam(required = true) int page, 
+			@RequestParam(required = true) int itemsPerPage, @RequestBody ProcessDataFilterParams filterProcess) {
 
 		//TODO this should be replaced by call to ControlService which is not yet introduced
 		Optional<Control> controlTableData = controlRepository.findById(controlTableId);
 
 		if (controlTableData.isEmpty()) {
 			return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ResponseMessage("success", 404,
-					"Process Data not found for control_id: " + controlTableId, "[]"));
+					"Process Data not found for control_id: " + controlTableId, null, "[]"));
+		} else if (page < 1) {
```

Comment:
ideally, we should use jakarta validations @min, @max etc. instead of manually doing in the code.

Reply:
The first check was originally in place and I just followed it.  I have not used jakarta validations before and I am not sure how to set up the min and max.

Reply:
it is ok as of now since it is just one field. You can google "spring+input request validation+ jakarta" you will get lot of examples. you can take a look at ColumnsDisplayPreference class and attribute displayName on how it is done.